File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> check;
        for(int i=0;i<nums.size();i++){
            if(check.find(target-nums[i])==check.end()){
                check[nums[i]]=i;
            }
            else{
                return {check[target-nums[i]],i};
            }
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        check = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in check:
                return [check[complement], i]
            else:
                check[num] = i
        
        return []